server/lorax_server/utils/sources/__init__.py

```diff
@@ -7,6 +11,21 @@
 HUB = "hub"
 S3 = "s3"
 LOCAL = "local"
+PBASE = "pbase"
+
+PREDIBASE_API_TOKEN = os.getenv("PREDIBASE_API_TOKEN", None)
+PREDIBASE_MODEL_URL_ENDPOINT = "/v1/models/version/name/{}?version={}"
+PREDIBASE_GATEWAY_ENDPOINT = os.getenv("PREDIBASE_GATEWAY_ENDPOINT", "https://api.predibase.com")
+
+@lru_cache(maxsize=256)
+def map_pbase_model_id_to_s3(model_id: str) -> str:
+    name, version = model_id.split("/")
+    url = PREDIBASE_GATEWAY_ENDPOINT + PREDIBASE_MODEL_URL_ENDPOINT.format(name, version)
+    headers = {"Authorization": f"Bearer {PREDIBASE_API_TOKEN}"}
+    resp = requests.get(url, headers=headers)
+    resp.raise_for_status()
+    uuid, best_run_id = resp.json()["uuid"], resp.json()["bestRunID"]
+    return f"{uuid}/{best_run_id}/artifacts/model/model_weights/"
```

Comment:
This feels a little too coupled to backend details. Can we have an endpoint that returns the full s3 URI: `s3://bucket/.../model_weights/`? I realize the current s3 source is coupled to an environment variable that specifies the bucket, but ideally we should change that, too (it makes it less useful for OSS users who want to use s3).

Reply:
Good point! Will fix now

Reply:
Okay, it's not a release blocker, but it is exposing a lot of unnecessary detail about our underlying storage system that will break if we ever move files around or want to deploy in Azure, etc. So definitely something we need to revisit.